homeassistant/components/camera/__init__.py

```diff
@@ -0,0 +1,1133 @@
+
+        self.hass.bus.listen(
+            EVENT_FTP_FILE_RECEIVED,
+            self.process_file_event)
+
+    def add_child_component_listeners(self):
+        """ Adds the child entity creation callback listeners """
+        self.hass.bus.listen(
+            self.entity_id + EVENT_CALLBACK_MOTION,
+            self.process_motion_switch_creation)
+
+        self.hass.bus.listen(
+            self.entity_id + EVENT_CALLBACK_RECORD,
+            self.process_child_switch_creation)
+
+        self.hass.bus.listen(
+            self.entity_id + EVENT_CALLBACK_SNAPSHOT,
+            self.process_child_switch_creation)
+
+    def process_motion_switch_creation(self, event):
+        """ Called when a child motion detection switch is created """
+        self.process_child_switch_creation(event)
+        self.send_motion_state()
+
+    def process_child_switch_creation(self, event):
+        """ Called when a child switch is created """
+        if not event or not event.data:
+            return
+
+        child_entity_id = event.data.get('entity_id')
+        entity_action = event.data.get('parent_action')
+        self._action_entities[entity_action] = child_entity_id
+
+        self._logger.info(
+            'Registerd child switch %s for %s on %s',
+            child_entity_id,
+            entity_action,
+            self.entity_id)
+
+    def send_motion_state(self):
+        """ Sends an event notifying listeners of the motion
+        detection state """
+        state = STATE_OFF
+        if self.is_motion_detection_enabled:
+            state = STATE_ON
+
+        self.hass.bus.fire(
+            self.entity_id + EVENT_CHANGE_MOTION,
+            {
+                'entity_id': self.entity_id,
+                'state': state
+            })
+
+    def send_recording_state(self):
+        """ Sends an event notifying listeners of the
+        recording state """
+        state = STATE_OFF
+        if self.is_recording:
+            state = STATE_ON
+
+        self.hass.bus.fire(
+            self.entity_id + EVENT_CHANGE_RECORD,
+            {
+                'entity_id': self.entity_id,
+                'state': state
+            })
+
+    def send_snapshot_state(self):
+        """ Sends an event notifying listeners of the
+        snapshot state """
+        state = STATE_OFF
+        if self.is_taking_snapshot:
+            state = STATE_ON
+
+        self.hass.bus.fire(
+            self.entity_id + EVENT_CHANGE_SNAPSHOT,
+            {
+                'entity_id': self.entity_id,
+                'state': state
+            })
+
+    def update(self):
+        """ Retrieve latest state. """
+        self.refesh_all_settings_from_device()
+        self.send_motion_state()
+
+    def refesh_all_settings_from_device(self):
+        """ A stub methos that should be overridden in derived classes to
+            fetch the settings from the camera. """
+        pass
+
+    def get_camera_image(self, stream=False):
```

Comment:
It seems that the stream parameter is never used. If that is the case, it would be nicer if this method would return `response.content` instead of `response`.